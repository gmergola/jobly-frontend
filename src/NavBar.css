.navbar {
  border-bottom: 8px solid rgb(75, 86, 139);
  opacity: 0.6;
  margin-bottom: 15px;
}

.navbar-jobly {
  font-size: 23px;
  color: black;
}

.navbar-links{
  font-size: 15px;
  color: black;
}

.NavBar-j {
  height: 40px;
  width: 40px;
}

a.active {
  color: rgb(75, 86, 139);
}

a:hover {
  opacity: 0.5;
}


@media only screen and (max-width: 767px) {
  .NavBar-j {
    height: 10px;
    width: 10px;
  }

  .navbar-jobly {
    font-size: 12px;
  }
  .navbar-links{
    font-size: 7px;
  }
}






