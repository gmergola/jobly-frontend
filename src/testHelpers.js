import React from "react";
import TokenContext from "./TokenContext";

const demoToken = 'TOKEN';

const TokenProvider = ({ children, currentToken = demoToken }) => (
  <TokenContext.Provider value={{currentToken}}>
    {children}
  </TokenContext.Provider>
);

export { TokenProvider };
