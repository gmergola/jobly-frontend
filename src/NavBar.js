import React, { useContext } from 'react';
import { NavLink } from 'react-router-dom';
import "./NavBar.css";
import TokenContext from "./TokenContext";
import { useHistory } from "react-router-dom";

/*NavBar: renders navbar on every page, depending on logged in status of user*/
function NavBar({ currentUser, setCurrentUser }) {
  const { setToken } = useContext(TokenContext);
  const history = useHistory();

  // upon logout, clears localStorage, resets token state (context) and redirects
  //back to homepage via history
  function handleLogout() {
    window.localStorage.clear();
    setToken("");
    setCurrentUser({});
    console.log("localStorage is:", localStorage);
    history.push('/');
  }

  //authenticates which navbar to render based on token state
  if (currentUser.username) {
    return (
      <nav>
        <ul>
          <li><NavLink exact to="/"><b>Home</b></NavLink></li>
          <li><NavLink exact to="/companies"><b>Companies</b></NavLink></li>
          <li><NavLink exact to="/jobs"><b>Jobs</b></NavLink></li>
          <li><NavLink exact to="/profile"><b>Profile</b></NavLink></li>
          <li><NavLink exact to="/" onClick={handleLogout} ><b>LogOut</b></NavLink></li>
        </ul>
      </nav>
    )
  } else {
    return (
      <nav>
        <ul>
          <li><NavLink exact to="/"><b>Home</b></NavLink></li>
          <li><NavLink exact to="/login"><b>Log In</b></NavLink></li>
        </ul>
      </nav>
    )
  }

}

export default NavBar;