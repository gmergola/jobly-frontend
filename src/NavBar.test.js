import React from "react";
import { render } from "@testing-library/react";
import NavBar from "./NavBar";
import { MemoryRouter } from "react-router";
import { TokenProvider } from "./testHelpers";


it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <TokenProvider>
        <NavBar />
      </TokenProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <TokenProvider>
        <NavBar />
      </TokenProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});