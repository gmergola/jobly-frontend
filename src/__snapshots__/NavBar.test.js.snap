// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <b>
            Home
          </b>
        </a>
      </li>
      <li>
        <a
          href="/login"
        >
          <b>
            Log In
          </b>
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
